### PR TITLE
feat(yarn): set http proxy config from environment variables

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Runs yarn global config commands to set http(s) proxy if the environment variables are set.

## Context

Yarn does not recognize the environment variables natively

## Documentation (please check one with an [x])

- [=x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/npm/readme.md [hunk 1]
```diff
@@ -7,3 +7,15 @@ The following `depTypes` are currently supported by the npm manager :
...
+
+#### Version Selection / Installation
+
+If Renovate detects a `packageManager` setting for Yarn in `package.json` then it will use Corepack to install Yarn.
+
+#### HTTP Proxy Support
+
+Yarn itself does not natively recognize/support the `HTTP_PROXY` and `HTTPS_PROXY` environment variables.
+If Renovate detects Yarn 2+, and one or both of those variables are present, then it will run commands like `yarn config set --home httpProxy http://proxy` prior to executing `yarn install`.
+This will result in the `~/.yarnrc.yml` file being created or modified with these settings, and the settings are not removed afterwards.
```
Comment (rarkins) [inline lib/modules/manager/npm/readme.md:21]: Is there any scenario where this is a problem?

Comment (viceice) [reply lib/modules/manager/npm/readme.md:21]: 🤔 maybe not sure

**Review by viceice — APPROVED** [review]

**Comment by lukeapage** [comment]
This breaks our setup because we want a proxy for renovate, but we don't want a proxy for yarn because its using internal servers. and because yarn doesn't support no-proxy, we can't tell yarn not to use a proxy :/

It seems ridiculous but would you be open to a config option that stops this feature happening?

**Comment by viceice** [comment]
please open a new discussion and reference this PR and the corresponding issue.

**Comment by fgreinacher** [comment]
> This breaks our setup because we want a proxy for renovate, but we don't want a proxy for yarn because its using internal servers. and because yarn doesn't support no-proxy, we can't tell yarn not to use a proxy :/
> 
> It seems ridiculous but would you be open to a config option that stops this feature happening?

This discussion continues at https://github.com/renovatebot/renovate/discussions/27983.

**Pull request merged**